graph-commons/src/main/scala/io/renku/data/ErrorMessage.scala

```diff
@@ -18,36 +18,59 @@
 
 package io.renku.data
 
+import cats.Show
 import eu.timepit.refined.api.{RefType, Refined}
 import eu.timepit.refined.collection.NonEmpty
+import io.circe.Json
 
 import java.io.{PrintWriter, StringWriter}
 
+sealed trait ErrorMessage {
+  type T
+  val value: T
+  def show: String
```

Comment:
I'm wondering if it is necessary to have that string representation - I thought it is only used to create responses?

Reply:
I think I answered your question in the comment above.